### DISCUSSION TITLE
Issue with arbitrary type serialization with namespaces

### DISCUSSION THREAD

**Discussion by xeru98** [nlohmann/json#3538] [Q&A] [answered]
Hey there I have a type A that is defined in the global namespace. So is to_json(). However I’m trying to call it from a function in a different namespace and it’s complaining. Any clue how to fix this

Global namespace:
class A{};
void to_json(json& j, const A& a);

function
B::C::Func(){
    A* a = getA();
    json j = *a; //this is claiming there is no suitable function
}

**Comment by gregmarr** [comment, marked as answer]
It works fine in gcc trunk: https://godbolt.org/z/P87cr4snf.

What compiler are you using, and what is the full error output?

**Reply by nlohmann** [reply]
@xeru98 Do you need further assistance?
